File: golang/1.两数之和.go
// package main

import (
	"fmt"
	"sort"
)

/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) (re []int) {
	if len(nums) < 2{
		return
	}
	m := make(map[int]int, len(nums))
	for i:=0; i<len(nums); i++ {
		cur_v := nums[i]
		cur_tar := target - cur_v
		if v, ok := m[cur_tar]; ok {
			re = append(re, i, v)
			sort.Ints(re)
			return
		} else {
			m[cur_v] = i
		}
	}
	return
}
// @lc code=end

func main()  {
	nums := []int{12, 7, 11, 15}
	target := 19
	fmt.Println(twoSum(nums, target))
}

File: py/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python
#
# [1] 两数之和
#

# @lc code=start
class Solution(object):
    def twoSum(self, nums, target):
        if len(nums) < 2:
            return []
        m = {}
        res = []
        for i in range(len(nums)):
            cur_v = nums[i]
            cur_tar = target - cur_v
            if cur_tar in m:
                res = [m.get(cur_tar), i]
                res.sort()
                return res
            else:
                m[cur_v] = i
        return res


        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
# @lc code=end


nums = [2, 7, 11, 15]
target = 9
print(Solution().twoSum(nums, target))